### PR TITLE
Update buttons in wxAuiPaneInfo when state changes

### PR DESCRIPTION
**Pull request by Kvaz1r**
Fixes for https://trac.wxwidgets.org/ticket/18223

If this changes are appropriate (I'm not really familiar with wxAUI) I also will add such updates for Maximize and Pin buttons.

### CODE REVIEW

**Comment by vadz** [comment]
Thanks, this looks reasonable to me, but I don't really know this code. E.g. I wonder why do we need to have `wxAuiPaneButtonArray buttons` at all, as it seems completely redundant if the buttons presence is defined just by the flags. Maybe we could simply get rid of it entirely, what do you think?

**Comment by Kvaz1r** [comment]
> E.g. I wonder why do we need to have `wxAuiPaneButtonArray buttons` at all, as it seems completely redundant if the buttons presence is defined just by the flags. Maybe we could simply get rid of it entirely, what do you think?

As far I understand `wxAuiPaneButtonArray buttons` need only for safely transfering pointer into [button field of wxAuiDockUIPart](https://github.com/wxWidgets/wxWidgets/blob/909adbb6bf832ada3db034cffc5548b8309a1602/include/wx/aui/framemanager.h#L733) class in [layouting](https://github.com/wxWidgets/wxWidgets/blob/04949050b2800dfb80c4609d57a51b2e0aff5198/src/aui/framemanager.cpp#L1831). So it makes sense. But it also possible replace pointer in field into just `int` and use `0` instead `NULL`. If do this way then class [`wxAuiPaneButton`](https://github.com/wxWidgets/wxWidgets/blob/909adbb6bf832ada3db034cffc5548b8309a1602/include/wx/aui/framemanager.h#L740) could be removed.

**Comment by vadz** [comment]
Yes, I'd like to get rid of this, I have no idea why do we need to use a pointer to `wxAuiPaneButton` rather than just a simple int containing `button_id`. If you feel like changing this, please do. TIA!

File: src/aui/framemanager.cpp [hunk 1]
```diff
@@ -1807,32 +1785,45 @@ void wxAuiManager::LayoutAddPane(wxSizer* cont,
...
         part.sizer_item = sizer_item;
         int caption_part_idx = uiparts.GetCount();
         uiparts.Add(part);
 
         // add pane buttons to the caption
-        int i, button_count;
-        for (i = 0, button_count = pane.buttons.GetCount();
-             i < button_count; ++i)
+        int button_count = 0;
+        const int N = 3;
```
Comment (vadz) [inline src/aui/framemanager.cpp]: This is really nitpicky, but I'd use `NUM_SUPPORTED_BUTTONS` or something like this instead, `N` is really too generic.

File: src/aui/framemanager.cpp [hunk 2]
```diff
@@ -1807,32 +1785,45 @@ void wxAuiManager::LayoutAddPane(wxSizer* cont,
...
+        wxAuiButtonId buttons[N] = {
+            wxAUI_BUTTON_MAXIMIZE_RESTORE,
+            wxAUI_BUTTON_PIN,
+            wxAUI_BUTTON_CLOSE
+        };
+        int flags[N] = {
+            wxAuiPaneInfo::buttonMaximize | wxAuiPaneInfo::buttonMinimize,
```
Comment (vadz) [inline src/aui/framemanager.cpp]: It looks like this is a change in behaviour: currently, the `MAXIMIZE_RESTORE` button seems to be only used if `HasMaximizeButton()` returns true, i.e. if `buttonMaximize` flag is set. Are you sure we also want to use it if `buttonMinimize` is?

**Review by vadz — COMMENTED** [review]
Thanks! This looks like a welcome simplification to me, but I wonder about the use of the restore button for minimize flag, could you please recheck this?

File: src/aui/framemanager.cpp (hunk 2 above)
Comment (Kvaz1r) [reply src/aui/framemanager.cpp]: No, it's my mistake - fixed now.

**Comment by vadz** [comment]
Thanks for fixing this!

**Pull request closed without merging**